webapp/channels/src/components/search_results/index.tsx

```diff
@@ -80,8 +81,15 @@ function makeMapStateToProps() {
         const currentSearch = (getCurrentSearchForSearchTeam(state) as unknown as Record<string, any>) || {};
         const currentTeamName = getCurrentTeam(state)?.name ?? '';
 
+        if (ownProps.isPinnedPosts) {
```

Comment:
This sorting needed to be moved out of the component render and into mapStateToProps, such that we sort the `Post` objects first.

I'm actually not sure if this sorting is needed, but I wanted to keep it to preserve the functionality in case it is.

Reply:
Looking back, this seems to date back to https://github.com/mattermost/mattermost-webapp/pull/297. I don't know why we didn't sort those results on the API (maybe we do now?) or if we still need this though